npk/src/manifest.rs

```diff
@@ -523,52 +566,135 @@ log:
         Ok(())
     }
 
+    /// Two mounts on the same target are invalid
+    #[tokio::test]
+    async fn duplicate_mount() -> Result<()> {
+        let manifest = "
+name: hello
+version: 0.0.0
+init: /binary
+mounts:
+  /dev: full 
+  /dev: full 
+";
+        assert!(Manifest::from_str(manifest).is_err());
+
+        Ok(())
+    }
+
+    #[test]
+    fn tmpfs() {
+        let manifest = "
+name: hello
+version: 0.0.0
+init: /binary
+mounts:
+  /a:
+    tmpfs: 100
+  /b:
+    tmpfs: 100k
+  /c:
+    tmpfs: 100m
+  /d:
+    tmpfs: 100g
+";
+        let manifest = Manifest::from_str(manifest).unwrap();
+        assert!(manifest.mounts.get(&PathBuf::from("/a")) == Some(&Mount::Tmpfs { size: 100 }));
+        assert!(
+            manifest.mounts.get(&PathBuf::from("/b")) == Some(&Mount::Tmpfs { size: 100 * 1024 })
+        );
+        assert!(
+            manifest.mounts.get(&PathBuf::from("/c"))
+                == Some(&Mount::Tmpfs {
+                    size: 100 * 1024 * 1024
+                })
+        );
+        assert!(
+            manifest.mounts.get(&PathBuf::from("/d"))
+                == Some(&Mount::Tmpfs {
+                    size: 100 * 1024 * 1024 * 1024
+                })
+        );
+
+        // Test a invalid tmpfs size string
+        let manifest = "
+name: hello
+version: 0.0.0
+init: /binary
+mounts:
+  /tmp:
+    tmpfs: 100M
+";
+        assert!(Manifest::from_str(manifest).is_err());
+    }
+
+    #[test]
+    fn invalid_dev() {
+        let manifest = "
+name: hello
+version: 0.0.0
+init: /binary
+mounts:
+  /dev:
+    tmpfs: 42
+";
+        assert!(Manifest::from_str(manifest).is_err());
+    }
+
+    #[test]
+    fn mount_ressource() {
+        let manifest = "
+name: hello
+version: 0.0.0
+init: /binary
+mounts:
+  /foo:
+    resource: foo-bar.qwerty12:0.0.1/
+";
+        Manifest::from_str(manifest).unwrap();
+    }
+
     #[test]
     fn serialize_back_and_forth() -> Result<()> {
         let m = "
 name: hello
 version: 0.0.0
 init: /binary
 args:
-    - one
-    - two
+  - one
+  - two
 env:
-    LD_LIBRARY_PATH: /lib
+  LD_LIBRARY_PATH: /lib
 mounts:
-    /lib:
-      host: /lib
-      flags:
-          - rw
-    /data: {}
-    /data_rw:
-      flags:
-          - rw
-    /here/we/go:
-        resource: bla:1.0.0/bin/foo
-    /tmpfs:
-        size: 42
-    /big_tmpfs:
-        size: 42G
```

Comment:
This last 2 lines should be reincorporated. The current test would break with this.

Reply:
Check for a later reformat commit of all the tests. The manifest strings are all aligned to two space indentation.
Tests are green ;-)

Reply:
I'm referring to the `tmpfs` with the unit letter. The test `serialize_back_and_forth` will break if this lines are kept, which is the reason why you removed them. I'm just not sure if this is a good idea.